src/internal/packager/compose.go

```diff
@@ -24,119 +27,115 @@ func GetComponents() (components []types.ZarfComponent) {
 	// Update the parent package config with the expanded sub components.
 	// This is important when the deploy package is created.
 	config.SetComponents(components)
-
-	return components
 }
 
-func GetComposedComponent(childComponent types.ZarfComponent) types.ZarfComponent {
-	message.Debugf("packager.GetComposedComponent(%+v)", childComponent)
+// GetComposedComponent recursively retrieves a composed zarf component
+func GetComposedComponent(parentComponent types.ZarfComponent) types.ZarfComponent {
+	message.Debugf("packager.GetComposedComponent(%+v)", parentComponent)
 
 	// Make sure the component we're trying to import cant be accessed
-	validateOrBail(&childComponent)
+	validateOrBail(&parentComponent)
```

Comment:
I know we discussed this a lot and went back and forth on the philosophy of imported/inherited relationships.  Since this is literally the opposite wording here, can we put some substantive comment/explanation since we've all been confused over this for so long and just changing from child->parent will be even more confusing for future devs.

Reply:
Added this:

```
// For composed components, we build the tree of components starting at the root and adding children as we go;
// this follows the composite design pattern outlined here: https://en.wikipedia.org/wiki/Composite_pattern
// where 1 component parent is made up of 0...n composite or leaf children.
```